libs/langchain/langchain/retrievers/self_query/redis.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from typing import Any, Tuple
+
+from langchain.chains.query_constructor.ir import (
+    Comparator,
+    Comparison,
+    Operation,
+    Operator,
+    StructuredQuery,
+    Visitor,
+)
+from langchain.vectorstores.redis import Redis
+from langchain.vectorstores.redis.filters import (
+    RedisFilterExpression,
+    RedisFilterField,
+    RedisFilterOperator,
+    RedisNum,
+    RedisTag,
+    RedisText,
+)
+from langchain.vectorstores.redis.schema import RedisModel
+
+_COMPARATOR_TO_BUILTIN_METHOD = {
+    Comparator.EQ: "__eq__",
+    Comparator.NE: "__ne__",
+    Comparator.LT: "__lt__",
+    Comparator.GT: "__gt__",
+    Comparator.LTE: "__le__",
+    Comparator.GTE: "__ge__",
+    Comparator.CONTAIN: "__eq__",
```

Comment:
Hmmm is this one right?

Reply:
i belieeeeve for tags == is what's used to check if a tag is in a list, but correct me if im wrong @Spartee

Reply:
Yes, ``==`` is the comparator.

We did this because ``!=`` also works for ``not in`` but also because ``__contains__`` (i.e. "in") works and ``not in`` doesn't. This is because we're overriding the ``__eq__`` behavior``. 

Originally we thought about just using the fluent style, but we've seen so many users mess that up in Redis-Py. It's also not very discoverable. 

Lots of other reasons too.

TLDR: Looks right.